Update documentation on darc (#24487)
dotnet/aspnetcore: docs/ReferenceResolution.md

  </ItemGroup>
```

## A darc cheatsheet

`darc` is a command-line tool that is used for dependency management in the dotnet ecosystem of repos. `darc` can be installed using the `darc-init` scripts located inside the `eng/common` directory. Once `darc` is installed, you'll need to set up the appropriate access tokens as outlined [in the official Darc docs](https://github.com/dotnet/arcade/blob/master/Documentation/Darc.md#setting-up-your-darc-client).

Once `darc` is installed and set-up, it can be used to modify the subscriptions and dependencies in a project.

**Getting the list of subscriptions in a repo**

Subscriptions are objects that define the ecosystem repos we are listening for updates to, the frequency we are looking for updates, and more.

> ```

Reviewer: nit: VS Code will complain about not specifying the type of code block. This is pretty generic so `bash` or `ps1` would both work. If you do this, remove the leading `$`s and do it for all the scripting you're adding.

$ darc get-subscriptions --target-branch master --target-repo aspnetcore$ --regex
```

**Disable/enable a subscription**

```
$ darc subscription-status --id {subscriptionIdHere} --enable
$ darc subscription-status --id {subscriptionIdHere} --disable
```

**Trigger a subscription**

Triggering a subscription will search for updates in its dependencies and open a PR in the target repo via the dotnet-maestro bot with these changes.

```
$ darc trigger-subscriptions --id {subscriptionIdHere}
```

**Manually update dependencies**

> If the `dotnet-maestro` bot has not correctly updated the dependencies, `darc update-dependencies` may be used to update the dependencies manually. Note, you'll need to run the commands below in a separate branch and submit a PR with the changes. These are the things that the bot would typically do for you if you use `trigger-subscriptions`.

Reviewer: nit
```suggestion
If the `dotnet-maestro` bot has not correctly updated the dependencies, `darc update-dependencies` may be used to update the dependencies manually. Note, you'll need to run the commands below in a separate branch and submit a PR with the changes. These are the things that the bot should do for you if you use `trigger-subscriptions` or automatically (when the subscription fires e.g. about 15 minutes after a dependency's build completes if `Update Frequency: EveryBuild`).
```

```
$ darc update-dependencies --channel '.NET Core 3.1 Release'

> $ darc update-depdendencies --channel '.NET 5 Dev' --source-repo efcore

Reviewer: ```suggestion
$ darc update-dependencies --channel '.NET 5 Dev' --source-repo efcore
```

```

Generally, using `trigger-subscriptions` is preferred for creating dependency updates instead of manually updating dependencies in your own PR.

**Toggling batchability of subscription**

> Subscriptions can be batched. When a dependency update is detected, darc will bundle the commits for that update with existing dependency PRs. To toggle whether a subscription is batched or not, you will need to use the `update-subscription` command.

Reviewer: ```suggestion
Subscriptions can be batched. When a dependency update is detected, `darc` will bundle the commits for that update with existing dependency PRs. To toggle whether a subscription is batched or not, you will need to use the `update-subscription` command.
```

```
$ darc update-subscription --id {subscriptionIdHere}
```

Your shell's default editor will open and allow you to edit the metadata of the subscription.

To disable batching, set `Batchable` to `False` and update the `Merge Policies` section with the following YAML.

```
  - Name: Standard
    Properties: {}
```

To enable batching, set `Batchable` to `True` and remove any `Merge Policies` set on the subscription.

Note: Merge policies can only be set on unbatched subscriptions. Be sure to set/unset the `Merge Policies` field properly as you toggle batchability.

> 

Reviewer: nit: Only need one final newline 😈
